test/basic/init_funcs.c

```diff
@@ -0,0 +1,21 @@
+#include "smack.h"
+
+// @expect verified
+// @flag --unroll=1
+
+void __SMACK_INIT(defineProcessStates) {
+  __SMACK_top_decl("const unique $process_uninitialized: int;");
```

Comment:
This is not an ideal usage of __SMACK_top_decl, since top_decl can be injected anywhere at it will still work. For that you don't really need __SMACK_INIT. Here is an example I had in mind:
int g = 10;

__SMACK_INIT(init1) {
  g = 11;
}

__SMACK_INIT(init2) {
  g = 12;
}

void main(void) {
  assert(g == 12);
}

Then in the _fail version, just change this assertion into:
  assert(g == 0 || g == 10 || g == 11);

I think that will be a better example...